### PR TITLE
[Core] Adding new test looking for error in #8096

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Trying to find the error from #8096. I added a new test that works in OMP. I do not know how to trigger the problem. In theory, it appeared when the DOF was only in the MPC but not in the element. This small test does that but does not trigger the problem, any idea why? @emiroglu 

**Changelog**
- New MPC test

### CODE REVIEW

**Comment by emiroglu** [comment]
It could be because there is only one `LinearMasterSlaveConstraint` defined on the "orphan" slave node. The [loop](https://github.com/KratosMultiphysics/Kratos/blob/7f61ff8d43086adee6112bb140d32850649a2782/kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h#L1165-L1172) in `ApplyConstraints` is executed only once which does not allow another thread to try to access the matrix at the same time. In #8096 the slave node is a slave to multiple master nodes.

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp [hunk 1]
```diff
@@ -88,6 +88,10 @@ namespace Kratos
             NodeType::Pointer pnode2 = rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
             NodeType::Pointer pnode3 = rModelPart.CreateNewNode(3, 2.0, 0.0, 0.0);
 
+            if (AdditionalNode) {
+                rModelPart.CreateNewNode(4, 3.0, 0.0, 0.0);
```
Comment (emiroglu) [inline kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:92]: rModelPart.CreateNewNode(5, 4.0, 0.0, 0.0);

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp [hunk 2]
```diff
@@ -127,6 +131,10 @@ namespace Kratos
 
             if (WithConstraint) {
                 rModelPart.CreateNewMasterSlaveConstraint("LinearMasterSlaveConstraint", 1, *pnode2, DISPLACEMENT_X, *pnode3, DISPLACEMENT_X, 1.0, 0.0);
+                if (AdditionalNode) {
+                    auto pnode4 = rModelPart.pGetNode(4);
+                    rModelPart.CreateNewMasterSlaveConstraint("LinearMasterSlaveConstraint", 2, *pnode3, DISPLACEMENT_X, *pnode4, DISPLACEMENT_X, 1.0, 0.0);
+                }
```
Comment (emiroglu) [inline kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:140]: auto pnode5 = rModelPart.pGetNode(5);
                    rModelPart.CreateNewMasterSlaveConstraint("LinearMasterSlaveConstraint", 3, *pnode3, DISPLACEMENT_X, *pnode5, DISPLACEMENT_X, 1.0, 0.0);

**Review by emiroglu — CHANGES_REQUESTED** [review]
This should trigger

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:92]: Oki doki

File: kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/tests/cpp_tests/strategies/builder_and_solvers/test_builder_and_solver.cpp:140]: Oki doki

**Review by emiroglu — APPROVED** [review]
OK from my side

**Comment by loumalouomega** [comment]
For some reason the RANS test are failing due to timeout... @sunethwarna

**Comment by philbucher** [comment]
I think the CI is just overall slow today (I haven't checked if github has any issues)
I also had to several times restart some runs

**Comment by sunethwarna** [comment]
Then I'll wait maybe untill tomorrow @loumalouomega . If this still persists, I'll check it. :) Would that be ok?

**Comment by loumalouomega** [comment]
> Then I'll wait maybe untill tomorrow @loumalouomega . If this still persists, I'll check it. :) Would that be ok?

False alarm

**Pull request merged**